dashboard/app/controllers/dynamic_config_controller.rb

```diff
@@ -65,4 +66,54 @@ def gatekeeper_set
     flash[:notice] = "Updated successfully! Remember your changes take 30 seconds to go into effect, so don't expect to see the changes immediately on this page."
     redirect_to action: :gatekeeper_show, feature: feature
   end
+
+  def dcdo_show
+    authorize! :read, :reports
+    DCDO.refresh
+
+    @dcdo_hsh = DCDO.to_h
+  end
+
+  def dcdo_set
+    authorize! :read, :reports
+    DCDO.refresh
+
+    params.require([:key, :data_type])
+    key = params[:key]
+    raw_value = params[:value]
+    data_type = params[:data_type]
+
+    begin
+      value =
+        case data_type
+        when "Integer"
+          Integer(raw_value)
+        when "Float"
+          Float(raw_value)
+        when "Boolean"
+          raw_value.to_bool
+        when "String"
+          raw_value
+        else
+          new_value = Oj.load(raw_value)
+          if new_value.class.to_s != data_type
+            raise "#{new_value} does not match data type \"#{data_type}\""
+          end
+          new_value
+        end
+
+      DCDO.set(key, value)
+
+      log_msg = "<b>DCDO - #{key}</b> #{current_user.name} set to #{value}"
```

Comment:
Will this start creating an _incomplete_ log of DCDO changes?  Do all DCDO set actions get logged this way (other than a direct edit in DynamoDB)?

Reply:
Dang. Yes. We copied the Gatekeeper web interface code, and brought over this functionality. I agree that the logging should be carried out by the underlying DCDO and Gatekeeper libraries. I think I want to save this for a future refactor of our dynamic configuration libraries.